src/App.tsx

```diff
@@ -1,61 +1,108 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/todo';
+import { User } from './types/user';
+
+function getUser(userId: number): User | null {
+  const foundedUser = usersFromServer.find(user => user.id === userId);
+
+  return foundedUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const users: User[] = usersFromServer.map(user => ({
+  ...user,
+}));
 
 export const App = () => {
+  const [userSelect, setUserSelect] = useState('');
+  const [titleInput, setTitleInput] = useState('');
+  const [isSubmit, setIsSubmit] = useState(false);
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+          setIsSubmit(true);
+          if (userSelect && titleInput) {
+            const selectedUser = users.find(user => user.name === userSelect);
+            const maxId = Math.max(...todos.map(todo => todo.id));
+            const todo:Todo = {
+              id: maxId + 1,
+              title: titleInput,
+              userId: selectedUser?.id || null,
+              user: selectedUser || null,
+              completed: false,
+            };
+
+            todos.push(todo);
+            setUserSelect('');
+            setTitleInput('');
+            setIsSubmit(false);
+          }
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">Title: </label>
+          <input
+            type="text"
+            id="titleInput"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={titleInput}
+            onChange={(event) => (setTitleInput(event.target.value))}
+          />
+          {(!titleInput && isSubmit)
```

Comment:
you can add anew variables for error cases and validate it in handle submit function